
Sort DHCP lease table in ascending IP order, capitalize MAC addresses, reformat MFR string, remove down icon, remove N/A text 
for pfSense 2.8.0

Espressomatic @UnraidForums - 7 June 2025


--- a/src/usr/local/www/status_dhcp_leases.php		2024-06-07 14:42:15
+++ b/src/usr/local/www/status_dhcp_leases.php		2024-06-07 18:45:28
@@ -131,6 +131,7 @@
 $expired_string = gettext("expired");
 $dynamic_string = gettext("dynamic");
 $static_string = gettext("static");
+$leases['lease']=array_reverse($leases['lease']);
 
 if ($_REQUEST['order']) {
 	usort($leases['lease'], function($a, $b) {
@@ -292,8 +293,7 @@
 			}
 		}
 	}
-
-	$mac = $data['mac'];
+	$mac = strtoupper($data['mac']);
 	$mac_hi = strtoupper($mac[0] . $mac[1] . $mac[3] . $mac[4] . $mac[6] . $mac[7]);
 ?>
 				<tr>
@@ -302,7 +302,7 @@
 <?php if ($data['online'] === $online_string): ?>
 						<i class="fa-solid fa-arrow-up text-success online" title="<?=htmlspecialchars($data['online'])?>"></i>
 <?php else: ?>
-						<i class="fa-solid fa-arrow-down online" title="<?=htmlspecialchars($data['online'])?>"></i>
+						<i class="fa"></i>
 <?php endif; ?>
 					</td>
 					<td><?=htmlspecialchars($data['ip'])?></td>
@@ -312,8 +312,7 @@
 					<td>
 <?php endif; ?>
 						<?=htmlspecialchars($mac)?>
-
-						<?php if (isset($mac_man[$mac_hi])):?>
+						<?php if (isset($mac_man[$mac_hi])):?></br>
 							(<?=htmlspecialchars($mac_man[$mac_hi])?>)
 						<?php endif; ?>
 					</td>
@@ -327,8 +326,8 @@
 						<td><?=htmlspecialchars($data['starts'])?></td>
 						<td><?=htmlspecialchars($data['ends'])?></td>
 					<? else: ?>
-						<td><?=gettext("n/a")?></td>
-						<td><?=gettext("n/a")?></td>
+						<td><?=gettext(" ")?></td>
+						<td><?=gettext(" ")?></td>
 					<? endif; ?>
 					<td>
 <?php if ($data['type'] == $dynamic_string): ?>
